k8s/kind/kind-cluster.sh

```diff
@@ -1,33 +1,91 @@
-#!/bin/sh
+#!/bin/bash
+
+set -euo pipefail
```

Comment:
Hi @sivaprasadreddy , Can you please check if this is working in mac, I have checked in Linux and windows